### PR TITLE
Fixes for using a plaintext password file

### PR DESCRIPTION
**Pull request by emoses**
I had trouble using the plaintext mqtt password option, so I fixed it.  See PR comments for details.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -200,12 +200,13 @@ async fn application_trampoline(config: &Config) -> Result<()> {
 
                 // It's not even an encrypted file, so we need to keep the permission settings pretty tight.
                 // The only time I can really enforce that is when reading the password.
-                if metadata.mode() == 0o600 {
+                if metadata.mode() & 0o777 == 0o600 {
```
Comment (emoses) [inline src/main.rs:203]: On the Debian system I'm using, `metadata.mode()` returns `0o100600`, where `0o100000` indicates it's a regular file (see https://www.gnu.org/software/libc/manual/html_node/Testing-File-Type.html)

File: src/main.rs [hunk 2]
```diff
@@ -200,12 +200,13 @@ async fn application_trampoline(config: &Config) -> Result<()> {
...
                     if metadata.uid() == users::get_current_uid() {
                         if metadata.gid() == users::get_current_gid() {
-                            fs::read_to_string(file_path)
+                            let pass: String = fs::read_to_string(file_path)
                                 .await
-                                .context("Failed to read password file.")?
+                                .context("Failed to read password file.")?;
+                            pass.as_str().trim_end().to_string()
```
Comment (emoses) [inline src/main.rs:209]: Trim a newline off the end of the file.

File: src/main.rs [hunk 3]
```diff
@@ -219,7 +220,7 @@ async fn application_trampoline(config: &Config) -> Result<()> {
         };
 
         client_builder.set_username(Some(username.into()));
-        client_builder.set_password(Some(password.into()));
+        client_builder.set_password(Some(password.as_bytes().to_vec()));
```
Comment (emoses) [inline src/main.rs:223]: I'm not sure this is necessary, I'm new to Rust :)

Comment (IamTheCarl) [reply src/main.rs:223]: There are several ways to have done it. This is perfectly fine.

**Review by IamTheCarl — APPROVED** [review]

**Pull request merged**